src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/IntermediateDateTypeConfiguratorTest.java

```diff
@@ -0,0 +1,602 @@
+    public static final String NEXT_HEARING_DATE_VALUE = GIVEN_DATE.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+        + "T16:00";
+
+    @Autowired
+    private DateTypeConfigurator dateTypeConfigurator;
+
+    @Test
+    public void should_calculate_based_on_intermediate_date_origin_in_origin_earliest_dates() {
+        ConfigurationDmnEvaluationResponse dueDateOriginEarliest = ConfigurationDmnEvaluationResponse.builder()
+            .name(stringValue("dueDateOriginEarliest"))
+            .value(stringValue("nextHearingDuration,nextHearingDate,priorityDate"))
+            .build();
+
+        ConfigurationDmnEvaluationResponse priorityDateOriginEarliest = ConfigurationDmnEvaluationResponse.builder()
+            .name(stringValue("priorityDateOriginEarliest"))
+            .value(stringValue("nextHearingDuration,nextHearingDate,dueDate"))
+            .build();
+
+        ConfigurationDmnEvaluationResponse calculatedDates = ConfigurationDmnEvaluationResponse.builder()
+            .name(stringValue("calculatedDates"))
+            .value(stringValue("nextHearingDate,nextHearingDuration,dueDate,priorityDate"))
+            .build();
+
+        ConfigurationDmnEvaluationResponse nextHearingDate = ConfigurationDmnEvaluationResponse.builder()
+            .name(stringValue("nextHearingDate"))
+            .value(stringValue(NEXT_HEARING_DATE_VALUE))
+            .build();
+
+        String intermediateDateName = "nextHearingDuration";
+        ConfigurationDmnEvaluationResponse nextHearingDurationOrigin = ConfigurationDmnEvaluationResponse.builder()
+            .name(stringValue(intermediateDateName + ORIGIN_SUFFIX))
+            .value(stringValue(INTERMEDIATE_DATE_VALUE))
+            .build();
+
+        List<ConfigurationDmnEvaluationResponse> evaluationResponses = readDueAndPriorityDateOriginFields(
+            intermediateDateName, dueDateOriginEarliest, priorityDateOriginEarliest, calculatedDates,
+            nextHearingDate, nextHearingDurationOrigin
+        );
+
+        List<ConfigurationDmnEvaluationResponse> configurationDmnEvaluationResponses
+            = dateTypeConfigurator.configureDates(evaluationResponses, false, false);
+
+        assertThat(configurationDmnEvaluationResponses)
+            .hasSize(5)
+            .isEqualTo(List.of(
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(stringValue("calculatedDates"))
+                    .value(stringValue("nextHearingDate,nextHearingDuration,dueDate,priorityDate"))
+                    .build(),
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(stringValue("nextHearingDate"))
+                    .value(stringValue("2022-10-13T16:00"))
+                    .build(),
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(stringValue("nextHearingDuration"))
+                    .value(stringValue("2022-10-19T21:00"))
```

Comment:
Should this "2022-10-19T21:00" be "2022-10-21T21:00" ? the set up is as below in builder
ConfigurationDmnEvaluationResponse.builder()
                .name(stringValue(intermediateDateName + NON_WORKING_DAYS_OF_WEEK_SUFFIX))
                .value(stringValue("SATURDAY,SUNDAY"))
                .build(),
            ConfigurationDmnEvaluationResponse.builder()
                .name(stringValue(intermediateDateName + SKIP_NON_WORKING_DAYS_SUFFIX))
                .value(stringValue("true"))
                .build(),

Reply:
https://github.com/hmcts/wa-task-management-api/blob/b83fc50a5c183ed9c79819781c324f85c08d2a02/src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/IntermediateDateTypeConfiguratorTest.java#L38-L39